backend/compact-connect/lambdas/cloudfront-csp/test/index.test.js

```diff
@@ -0,0 +1,388 @@
+//
+//  index.test.js
+//  CompactConnect
+//
+//  Created by InspiringApps on 7/22/2024.
+//
+
+const {
+    expect,
+    testFilename,
+    runLambda,
+    lambdaConfig
+} = require('./config');
+
+// ================================================================================================
+// =                                        SETUP                                                 =
+// ================================================================================================
+const environments = {
+    csg: {
+        prod: {
+            webFrontend: `app.compactconnect.org`,
+            dataApi: `api.compactconnect.org`,
+            s3Upload: ``,
+        },
+        test: {
+            webFrontend: `app.test.compactconnect.org`,
+            dataApi: `api.test.compactconnect.org`,
+            s3Upload: `test-persistentstack-mockbulkuploadsbucket0e8f27eb-ivslg22f2yfz.s3.amazonaws.com`,
+        },
+    },
+    ia: {
+        prod: {
+            webFrontend: `app.jcc.iaapi.io`,
+            dataApi: `api.jcc.iaapi.io`,
+            s3Upload: ``,
+        },
+        test: {
+            webFrontend: `app.test.jcc.iaapi.io`,
+            dataApi: `api.test.jcc.iaapi.io`,
+            s3Upload: `test-persistentstack-mockbulkuploadsbucket0e8f27eb-4h1anohxetmp.s3.amazonaws.com`,
+        },
+        justin: {
+            webFrontend: `app.justin.jcc.iaapi.io`,
+            dataApi: `api.test.jcc.iaapi.io`,
+            s3Upload: `test-persistentstack-mockbulkuploadsbucket0e8f27eb-4h1anohxetmp.s3.amazonaws.com`,
+        },
+    },
+};
+const buildCspHeaders = (environment) => {
+    const dataApiUrl = (environment?.dataApi) ? `https://${environment.dataApi}` : '';
+    const s3Url = (environment?.s3Upload) ? `https://${environment.s3Upload}` : '';
+    const cspDefaultSrc = [
+        '\'none\'',
+    ].join(' ');
+    const cspManifestSrc = [
+        '\'self\'',
+    ].join(' ');
+    const cspScriptSrc = [
+        '\'self\'',
+    ].join(' ');
+    const cspScriptSrcElem = [
+        '\'self\'',
+    ].join(' ');
+    const cspScriptSrcAttr = [
+        '\'self\'',
+    ].join(' ');
+    const cspWorkerSrc = [
+        '\'self\'',
+    ].join(' ');
+    const cspStyleSrc = [
+        '\'self\'',
+        'https://fonts.googleapis.com',
+    ].join(' ');
+    const cspStyleSrcElem = [
+        '\'self\'',
+        'https://fonts.googleapis.com',
+    ].join(' ');
+    const cspStyleSrcAttr = [
+        '\'self\'',
+    ].join(' ');
+    const cspFontSrc = [
+        '\'self\'',
+        'https://fonts.gstatic.com',
+    ].join(' ');
+    const cspImgSrc = [
+        '\'self\'',
+        'data:',
+        dataApiUrl,
```

Comment:
What's this doing?

Reply:
Sometimes SVG images have base64 values / bodies. In those case the `<img src` uses the `data:` scheme. It's oddly common with font & css libraries. Here we're restricting it to image sources.